tests/framework/client/node.go

```diff
@@ -0,0 +1,134 @@
+package client
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"kubevirt.io/kubevirt/tests/util"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	corev1 "k8s.io/client-go/applyconfigurations/core/v1"
+	v12 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+var createdNodes = make(map[string]nodeInfo, 0)
+var nodeSpecMap = make(map[string]nodeSpec, 0)
+
+type nodeSpec struct {
+	labels map[string]string
+	spec   v1.NodeSpec
+}
+
+type nodeInfo struct {
+	name string
+	ctx  context.Context
+}
+
+// nodes implements NodesInterface
+type nodes struct {
+	client rest.Interface
+	node   v12.NodeInterface
+}
+
+// newNodes returns a Nodes
+func newNodes(c *TestCoreV1Client) *nodes {
+	return &nodes{
+		client: c.core.RESTClient(),
+		node:   c.core.Nodes(),
+	}
+}
+
+func (c *nodes) Clean() {
+	for _, node := range createdNodes {
+		err := c.Delete(node.ctx, node.name, metav1.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			panic(err)
+		}
+	}
+	for name, node := range nodeSpecMap {
```

Comment:
We could also check if we need to update the node.

Reply:
I add a check on clean as you suggested, thanks